### ISSUE TITLE
Allowances - Revoke - UI

### ISSUE THREAD

**Issue by ezra-sg** [telosnetwork/telos-wallet#605]
blocked by #206 

Create a modal UI for editing the allowances for an asset

 - there should be a pencil icon next to the amounts in the table, which has proper a11y
- clicking the pencil icon in the table should launch a modal for that asset
    - modal should be its own component, something like `EditAllowanceModal`
- the modal title should be "Define new {symbol | asset name up to 20 characters} allowance for {spender contract - external link}"
- there should be three options, None Unlimited and Custom (all should be added to i18n)
- custom should have a currency input which is disabled unless Custom is selected
    - if you edit the token amount input, then click a different option like None, the token amount input should be cleared
- **TODO should there be any constraints on the currency input, like number of decimals?** limit to 999T, max decimals is decimals for contract. for NFTs, only integers allowed
- confirm should do nothing (implemented in #606)
- cancel and clicking outside the modal should close the modal

Update allowances table to include revoke checkboxes
- allowances table should get a new column for selecting rows
- clicking the checkbox in the header cell should select all visible rows
- the table controls at the foot of the table should be updated to include a new layout to allow for arbitrary text, like 'x rows selected' (see below image) 

![Image](https://github.com/telosnetwork/telos-wallet/assets/11837820/e83fbddb-a84e-44be-9a48-95922b973022)
